src/App.tsx

```diff
@@ -15,36 +16,104 @@
   'Garlic',
 ];
 
+enum SortType {
+  alph = 'alph',
+  length = 'length',
+}
+
+interface SortParams {
+  sortField: string | number,
+  reversed: boolean
+}
+
+const getPreparedGoods = (goods: string[], { sortField, reversed }: SortParams) => {
+  let preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.alph:
+          return good1.localeCompare(good2);
+        case SortType.length:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reversed) {
+    preparedGoods = preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+};
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [reversed, setReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    reversed,
+  });
+
+  const resetSortField = () => {
+    setSortField('');
```

Comment:
In the `resetSortField` function, `setSortField('')` sets `sortField` to an empty string, which might not be consistent with the `SortType` enum. Consider resetting it to a valid `SortType` value or handling the empty string case in your logic.